tests/rptest/services/redpanda.py

```diff
@@ -3527,8 +3527,8 @@ def metrics_samples(
         }
 
         for node in nodes:
-            metrics = self.metrics(node, metrics_endpoint)
             for pattern in sample_patterns:
+                metrics = self.metrics(node, metrics_endpoint)
```

Comment:
Wouldn't it be better to materialize the list so we're not doing multiple metrics scapes?

IE:

```
for node in nodes:
  # Materialize this generator so it can be used multiple times.
  metrics = [m for m in self.metrics(node, metrics_endpoint)]
  for pattern in sample_patterns:
    sample_values_per_pattern[pattern] += self._extract_samples(metrics, pattern, node)
```

Or maybe `self.metrics` should materialize the generator and return a list...

Reply:
Missed this...incidentally, I had it that way on another PR and the argument for keeping it a generator was roughly that the results here could be rather large. This facility is hardly used, so I can't say I have a strong opinion in this particular case. In general I'd lean towards materializing the list since we're in a test environment.

Reply:
Thanks for the response. It's fine to keep it, we're just trading memory for latency, and in general because as you note about the test environment (and it's python) I'd just save on the latency.

Reply:
I don't have a strong opinion here. I've argued to keep it a generator on about 3 PRs now and so it seems that it's inconvenient. Probably worth checking the git history to see if it ever caused a real problem because I've forgotten why I keep arguing for generator.